include/mqtt/endpoint.hpp

```diff
@@ -5946,17 +5946,30 @@ class endpoint : public std::enable_shared_from_this<endpoint<Socket, Mutex, Loc
     }
 
     void do_async_write() {
-        auto const& elem = queue_.front();
-        auto const& mv = elem.message();
+        std::vector<as::const_buffer> buf;
+        std::vector<async_handler_t> handlers;
```

Comment:
Don't forget std::vector<>::reserve()

Reply:
I think that there is no easy way to know the number of `as::const_buffer`.  See https://github.com/redboltz/mqtt_cpp/blob/master/include/mqtt/message.hpp#L741

In order to get the number of chunks, need to add the new member function such as `num_of_const_buffers()` to each message type. And call it via visitor mechanism.

Reply:
I will add by this approach.

Reply:
I added.
This approach requires one more loop for calculating the number of const buffer sequence. Call getting the number of const buffer sequence function for each element via visitor.

I'm not sure that this approach is really more efficient than allow vector expansion (reallocation).

Reply:
I believe that calling reserve approach is efficient even if additional loop and visitor call is executed. vector reallocation could invoke system calls. It is high cost.